Sources/FunctionalTableData/CellStyle.swift

```diff
@@ -151,12 +155,15 @@ public struct CellStyle {
 			cell.contentView.insetsLayoutMarginsFromSafeArea = false
 		}
 		
+		cell.selectedBackgroundView = nil
 		if let selectionColor = selectionColor {
 			let selectedBackgroundView = UIView()
 			selectedBackgroundView.backgroundColor = selectionColor
 			cell.selectedBackgroundView = selectedBackgroundView
-		} else {
-			cell.selectedBackgroundView = nil
+		}
+		
+		if let selectedBackgroundView = selectionBackgroundViewProvider?.backgroundView() {
+			cell.selectedBackgroundView = selectedBackgroundView
```

Comment:
nit: To be clear that you only want one, you could rearrange this to be:
```
if let selectedBackgroundView = selectionBackgroundViewProvider?.backgroundView() {
    cell.selectedBackgroundView = selectedBackgroundView
} else if let selectionColor = selectionColor {
    let selectedBackgroundView = UIView()
    selectedBackgroundView.backgroundColor = selectionColor
    cell.selectedBackgroundView = selectedBackgroundView
}
```
But I see why you followed the structure above, so I'm not strong on this.